mdp/nodes/misc_nodes.py

```diff
@@ -815,3 +826,603 @@ def _execute(self, x):
+        self._shape_buffer = self.input_shape
+
+        # output to 2d
+        x = x.reshape(x.shape[0], mdp.numx.product(x.shape[1:]))
+
+        return self._refcast(x)
+
+    # utility methods
+
+    def get_transform_fn_names(self):
+        """Returns the available transformation function names"""
+        fns = self._get_transform_fns()
+        return fns.keys()
+
+
+# TODO: Could use or add to the graph package of MDP?
+class GridProcessingNode(mdp.Node):
+    """
+    GridProcessingNode is a non-trainable node that
+    translates an n-dimensional cartesian grid to a topological map
+    of graph nodes labeled with n-dimensional coordinates and
+    a 1D index. Therefore, the node provides conversions
+    from grid coordinates to graph coordinates or graph indices.
+
+    The type of conversion can be selected by setting the 'output_type'
+    attribute. Accepted values:
+    'gridx' - default (just returns the input)
+    'graphx' - returns graph coordinates
+    'graphindx' - returns graph indices of the input.
+
+    The node is also invertible, enabling conversions
+    back from 'graphx' and 'graphindx' to 'gridx'.
+
+    Eg.
+
+    3x3 cartesian grid points to  'graphx' output_type
+
+    (0,0)   (2,0)   (4,0)       (0,0)   (1,0)   (2,0)
+
+    (0,2)   (2,2)   (4,2)  <->  (0,1)   (1,1)   (2,1)
+
+    (0,4)   (2,4)   (4,4)       (0,2)   (1,2)   (2,2)
+
+    3x3 cartesian grid points to 'graphindx' output_type
+
+    (0,0)   (2,0)   (4,0)       0   1   2
+
+    (0,2)   (2,2)   (4,2)  <->  3   4   5
+
+    (0,4)   (2,4)   (4,4)       6   7   8
+
+
+    The node also provides a utility method:
+     'get_neighbors' - Returns the neighboring
+     grid points based on a distance function applied to
+     the graph coordinates. The distance function
+     can be specified using the 'nbr_dist_fn' argument
+     and the threshold can be specified using the
+     'nbr_radius' argument.
+
+     'get_adjacency_matrix' - Returns an adjacency matrix
+     of the graph.
+
+    """
+
+    def __init__(self, grid_lims, n_grid_pts=None, output_type=None, nbr_dist_fn=1, nbr_radius=1,
+                 include_central_elem=True, input_dim=None, dtype=None):
+        """
+        grid_lims - a tuple of lower and upper bounds for each dimension of the input grid.
+                Eg., [(lower1, lower2, ...,), (upper1, upper2, ...,)]
+
+        n_grid_pts - number of grid points within each dimension. This provides the spacing between
+                the grid points. Default is set to the number of integers between the bounds.
+
+        output_type - the type of conversion. Accepted values:
+                    'gridx' - default (just returns the input)
+                    'graphx' - returns graph coordinates
+                    'graphindx' - returns graph indices of the input.
+
+        nbr_dist_fn - distance function to define the closest neighbors (applied on the graph
+                    coordinates). This is the same as
```

Comment:
I'd change the wording just a little: "This is passed as the "order" argument to the numx...".

Maybe we can improve the argument name a little, the "fn" ending sounds to me like this should be a callable. "nbr_norm_ord" for example would be closer to the argument name in NumPy.